docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+What if we want to access our network status data from another component? Since
+we don't know whether our `UPDATE_NETWORK_STATUS` mutation will fire before we
+try to access the data, we should guard against undefined values by providing an
+initial state for our query in our resolver map.
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
+  Query: {
+    networkStatus: () => null,
+  },
+});
+```
+
+How do we query the `networkStatus` from our component? Similar to mutations,
+just use a query & the `@client` directive! With Apollo Link, we can combine
+data sources, including your remote data, in one query.
+
+In this example, the `articles` field will either hit the cache or fetch from
+our GraphQL endpoint, depending on our fetch policy. Since `networkStatus` is
+marked with `@client`, we know that this is local data, so it will resolve from
+the cache.
+
+```js
+const GET_ARTICLES = gql`
+  query {
+    networkStatus @client {
+      isConnected
+    }
+    articles {
+      id
+      title
+    }
+  }
+`;
+```
+
+To retrieve the data in your component, bind your query to your component via
+your favorite Apollo view layer integration just like you normally would. Then,
+React Apollo will attach both your remote & local data to `props.data` while
+tracking both loading & error states. Once the query returns a result, your
+component will update reactively.
+
+```js
+const WrappedComponent = graphql(GET_ARTICLES, {
+  props: ({ data: { networkStatus, articles } }) => {
+    if (data.loading) {
+      return { loading: data.loading };
+    }
+
+    if (data.error) {
+      return { error: data.error };
+    }
+
+    return {
+      loading: false,
+      networkStatus,
+      articles,
+    };
+  },
+})(Articles);
+```
+
+Now that you've seen how easy it is to manage your local state in Apollo Client,
+let's dive deeper into how `apollo-link-state` updates & queries your local
+data: the resolver map.
+
+<h2 id="resolver">Resolver map</h2>
+
+Your resolver map is where all the magic happens to retrieve and update your
+local data in the Apollo cache. The resolver map is an object with resolver
+functions for each GraphQL object type. You can think of a GraphQL query or
+mutation as a tree of function calls for each field. These function calls
+resolve to data or another function call.
+
+The three most important things to keep in mind about resolvers in
+`apollo-link-state` are this:
+
+1. The cache is added to the context for you so you can write & read data from
```

Comment:
maybe be super explicit here: "the context (ie, the third argument to the resolver)"